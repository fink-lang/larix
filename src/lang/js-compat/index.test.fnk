{describe, it, expect, to_match_snapshot} = import '@fink/jest'

{parse_expr, parse} = import '../..'


describe 'js-compat operators', fn:

  it 'parses new', fn:
    expect
      parse_expr 'new_foo = new Foobar(spam, ni)'
      to_match_snapshot

    expect
      parse_expr 'new_foo = new Foobar spam, ni'
      to_match_snapshot


  it 'parses throw', fn:
    expect
      parse_expr `throw_foo = foo or throw err('foobar')`
      to_match_snapshot

    expect
      parse_expr `throw_foo = foo or throw err 'foobar'`
      to_match_snapshot


describe 'try ...:', fn:

  it 'parses', fn:
    expect
      parse `
        [err, result] = try:
          throw new Error 'foobar'
      `
      to_match_snapshot



describe 'constants', fn:

  it 'parses null', fn:
    expect
      parse `
        foo = null
        bar = nullish
      `
      to_match_snapshot
