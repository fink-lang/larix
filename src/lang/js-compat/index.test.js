import {parse_expr} from '../..';


describe('js-compat operators', ()=> {
  it('parses new', ()=> {
    expect(
      parse_expr('new Foobar(spam, ni)')
    ).toMatchSnapshot();
  });

  it('parses throw', ()=> {
    expect(
      parse_expr('throw err(`foobar`)')
    ).toMatchSnapshot();
  });
});
