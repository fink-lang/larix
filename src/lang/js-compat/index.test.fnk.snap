// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attempt ...: parses 1`] = `
module (1:0-3:0)
  assign = (1:0-3:0)
    array (1:0-1:13)
      ident (1:1-1:4) err
      ident (1:6-1:12) result
    js-compat attempt (1:16-3:0)
      ident (1:24-1:28) item
      :
      js-compat throw (2:2-3:0)
        js-compat new (2:8-3:0)
          call (2:12-3:0)
            ident (2:12-2:17) Error
            string ' (2:18-2:26)
              'foobar'
`;

exports[`js-compat operators parses new 1`] = `
assign = (1:0-1:30)
  ident (1:0-1:7) new_foo
  js-compat new (1:10-1:30)
    call ( (1:14-1:30)
      ident (1:14-1:20) Foobar
      ident (1:21-1:25) spam
      ident (1:27-1:29) ni
`;

exports[`js-compat operators parses new 2`] = `
assign = (1:0-1:31)
  ident (1:0-1:7) new_foo
  js-compat new (1:10-1:31)
    call :: (1:14-1:31)
      ident (1:14-1:20) Foobar
      ident (1:23-1:27) spam
      ident (1:29-1:31) ni
`;

exports[`js-compat operators parses new 3`] = `
assign = (1:0-1:29)
  ident (1:0-1:7) new_foo
  js-compat new (1:10-1:29)
    call (1:14-1:29)
      ident (1:14-1:20) Foobar
      ident (1:21-1:25) spam
      ident (1:27-1:29) ni
`;

exports[`js-compat operators parses throw 1`] = `
assign = (1:0-1:38)
  ident (1:0-1:9) throw_foo
  logical || (1:12-1:38)
    ident (1:12-1:15) foo
    js-compat throw (1:19-1:38)
      call ( (1:25-1:38)
        ident (1:25-1:28) err
        string ' (1:29-1:37)
          'foobar'
`;

exports[`js-compat operators parses throw 2`] = `
assign = (1:0-1:39)
  ident (1:0-1:9) throw_foo
  logical || (1:12-1:39)
    ident (1:12-1:15) foo
    js-compat throw (1:19-1:39)
      call :: (1:25-1:39)
        ident (1:25-1:28) err
        string ' (1:31-1:39)
          'foobar'
`;

exports[`js-compat operators parses throw 3`] = `
assign = (1:0-1:37)
  ident (1:0-1:9) throw_foo
  logical || (1:12-1:37)
    ident (1:12-1:15) foo
    js-compat throw (1:19-1:37)
      call (1:25-1:37)
        ident (1:25-1:28) err
        string ' (1:29-1:37)
          'foobar'
`;
